R/mod_review_forms.R

```diff
@@ -241,46 +235,40 @@ mod_review_forms_server <- function(
           status      = if(input$form_reviewed) "old" else "new"
         ) 
       
-      golem::cat_dev("review row to add:\n")
-      golem::print_dev(review_row)
+      golem::cat_dev("review records to add:\n")
+      golem::print_dev(review_records)
       
       cat("write review progress to database\n")
       db_save_review(
-        review_row, 
+        review_records, 
         db_path = db_path,
-        # More tables can be added here if needed, to track process of 
-        # individual reviewers in individual tables:
-        tables = "all_review_data" 
+        table = "all_review_data" 
       )
       
-      # Contains multiple rows, one for each item.
       updated_rows_db <- db_get_review(
-        db_path, subject = review_row$subject_id, form = review_row$item_group
-        )[c(names(review_row), "event_name", "item_name", "form_repeat")]
-      # Within a form, only items with a changed review state are updated and 
-      # contain the new (current) time stamp. 
-      updated_rows_db <- updated_rows_db[
-        updated_rows_db$timestamp == review_row$timestamp[1], 
-        ]
+        db_path, id = review_records$id
+        ) |> 
+        dplyr::select(dplyr::all_of(names(review_records)))
       
-      review_row_db <- unique(updated_rows_db[names(review_row)])
-      if(identical(review_row_db, review_row)){
+      review_records_db <- updated_rows_db |> 
+        # Within a form, only items with a changed review state are updated and 
+        # contain the new (current) time stamp. 
+        dplyr::filter(timestamp == review_records$timestamp[1])
+      if(isTRUE(all.equal(review_records_db, review_records, check.attributes = FALSE))){
```

Comment:
The issue is that this statement now fails: the data frame `review_records` contains all possible item updates, but in the database only records with a changed review status will be updated. Thus, this statement does not always return `TRUE`